File: KILLKTH.c
#include<iostream.h>
#include<string.h>
using namespace std
int main()
{
    string a,c;
    cin>>a;
    vector<sring> list;
    int len=strlen(a);
    int up=0,down=0;
    while(up<n)
    {
        while(down<n)
        {
        char g[10000];
        int l=0;
        for(int i=up;i<=down;i++)
            g[l++]=a[i];
        g[l]='\0';
        list.push_back(g);
        down++;
        }
        down=up+1;
        up++;
    }
    list.sort();
    int x=list.size();
    c=list[0];
    for(int i=1;i<x;i++)
        c+=list[i];
    int q,m;
    long int p;
    long int g=0;
    scanf("%d",&q);
    while(q--)
    {
        scanf("%ld%d",&p,&m);
        int y=(p*g)%m;
        printf("%c\n",c[y]);
    }
    return 0;
}

File: KILLKTH.py
s=input();
x=len(s);
lis=[];
up=0;
down=1;
while up<x:
  while down<=x:
    g=s[up:down];
    lis.append(g);
    down=down+1;
  up=up+1;
  down=up+1;
lis.sort();
c=''.join(lis);
m=int(input());
h=0;
for i in range(m):
  p=int(input());
  m=int(input());
  y=((p*g)%m);
  print(c[y]);
  g+=c[y];

